File: artifacts/tulipindicators/translate_without_break_down/src/obv.rs
use crate::*;

pub fn ti_obv_start(options: Option<&[f64]>) -> i32 {
    let _ = options; // Explicitly ignore the parameter to match C behavior
    0
}
pub fn ti_obv(
    size: i32,
    inputs: &[&[f64]],
    options: &[f64],
    outputs: &mut [&mut [f64]],
) -> i32 {
    let close = inputs[0];
    let volume = inputs[1];
    let _ = options; // Explicitly ignore options to avoid unused variable warning
    let output = &mut outputs[0];

    let mut output_idx = 0;
    let mut sum = 0.0;
    output[output_idx] = sum;
    output_idx += 1;

    let mut prev = close[0];
    for i in 1..size as usize {
        if close[i] > prev {
            sum += volume[i];
        } else if close[i] < prev {
            sum -= volume[i];
        }
        prev = close[i];
        output[output_idx] = sum;
        output_idx += 1;
    }

    0
}

File: artifacts/tulipindicators/decompose/src/obv.c
int ti_obv_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_obv(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *close = inputs[0];
  const double *volume = inputs[1];
  (void) options;
  double *output = outputs[0];
  unsigned int output_idx = 0;
  double sum = 0;
  output[output_idx] = sum;
  output_idx += 1;
  double prev = close[0];
  int i;
  for (i = 1; i < size; i += 1)
  {
    if (close[i] > prev)
    {
      sum += volume[i];
    }
    else
      if (close[i] < prev)
    {
      sum -= volume[i];
    }
    else
    {
    }
    prev = close[i];
    output[output_idx] = sum;
    output_idx += 1;
  }

  return 0;
}


/*** DEPENDENCIES:
***/


